apis/tempo/v1alpha1/microservices_webhook.go

```diff
@@ -192,6 +211,23 @@ func (v *validator) validateStorage(ctx context.Context, tempo *Microservices) f
 	return v.validateStorageSecret(tempo, storageSecret)
 }
 
+func (v *validator) validateReplicationFactor(tempo *Microservices) field.ErrorList {
+	// Validate minimum quorum on ingestors according to replicas and replication factor
+	replicatonFactor := tempo.Spec.ReplicationFactor
+	ingesterReplicas := tempo.Spec.Components.Ingester.Replicas
+	quorum := int32(math.Floor(float64(replicatonFactor)/2.0) + 1)
+	// if ingester replicas less than quorum (which depends on replication factor), then doesn't allow to deploy as it is an
+	// invalid configuration. Quorum equal to replicas doesn't allow you to lose ingesters but is a valid configuration.
+	if (ingesterReplicas == nil && quorum > 1) || *ingesterReplicas < quorum {
```

Comment:
If the default webhook is run, `tempo.Spec.Components.Ingester.Replicas` can never be `nil` here, right?
I'm wondering, is it possible that the default webhook doesn't run but the validation webhook runs?

Reply:
No, I did this only because double check and precautions, but I'll change those things to not be pointers.